test/e2e/prune_test.go

```diff
@@ -580,4 +588,69 @@ var _ = Describe("Podman prune", func() {
 		Expect(err).ToNot(HaveOccurred())
 		Expect(dirents).To(HaveLen(3))
 	})
+
+	It("podman system prune --build clean up after terminated build", func() {
+		useCustomNetworkDir(podmanTest, tempdir)
+		containerStorageDir := filepath.Join(podmanTest.Root, podmanTest.ImageCacheFS+"-containers")
+
+		podmanTest.BuildImage(pruneImage, "alpine_notleaker:latest", "false")
+
+		create := podmanTest.Podman([]string{"create", "--name", "test", BB})
+		create.WaitWithDefaultTimeout()
+		Expect(create).Should(ExitCleanly())
+
+		containerFilePath := filepath.Join(podmanTest.TempDir, "ContainerFile-"+stringid.GenerateRandomID())
+		err := os.WriteFile(containerFilePath, []byte(longBuildImage), 0755)
+		Expect(err).ToNot(HaveOccurred())
+
+		build := podmanTest.Podman([]string{"build", "-f", containerFilePath, "-t", "podmanleaker"})
+		// Build will never finish so let's kill it after 8 seconds to simulate a failed build that leaves stage containers
+		time.Sleep(8 * time.Second)
+		build.Signal(syscall.SIGKILL)
+
+		count := podmanTest.Podman([]string{"ps", "-aq", "--external"})
+		count.WaitWithDefaultTimeout()
+		Expect(count).Should(ExitCleanly())
+		Expect(count.OutputToStringArray()).To(HaveLen(3))
+		Expect(podmanTest.NumberOfContainers()).To(Equal(1))
+
+		// Check Intermediate image of stage container
+		none := podmanTest.Podman([]string{"images", "-a"})
+		none.WaitWithDefaultTimeout()
+		Expect(none).Should(ExitCleanly())
+		hasNone, result := none.GrepString("<none>")
+		Expect(result).To(HaveLen(1))
+		Expect(hasNone).To(BeTrue())
+
+		// have: volatile-containers.json, containers.json, containers.lock and container dir and 2 stage container dirs
+		dirents, err := os.ReadDir(containerStorageDir)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(dirents).To(HaveLen(6))
+
+		prune := podmanTest.Podman([]string{"system", "prune", "--build", "-f"})
+		prune.WaitWithDefaultTimeout()
+		Expect(prune).Should(ExitCleanly())
+
+		// Container should still exist, but no stage containers
+		count = podmanTest.Podman([]string{"ps", "-aq", "--external"})
+		count.WaitWithDefaultTimeout()
+		Expect(count).Should(ExitCleanly())
+		Expect(count.OutputToStringArray()).To(HaveLen(1))
+		Expect(podmanTest.NumberOfContainers()).To(Equal(1))
+
+		after := podmanTest.Podman([]string{"images", "-a"})
+		after.WaitWithDefaultTimeout()
+		Expect(after).Should(ExitCleanly())
+		Expect(len(after.OutputToStringArray())).To(BeNumerically(">", 1))
```

Comment:
Not sure what this should check, AFAIK in the test/e2e setup we will always have extra images from the additional store shown.